### PR TITLE
Update to edition 2021

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: sixtyfps_runtime/rendering_backends/qt/qt_window.rs [hunk 1]
```diff
@@ -497,7 +497,7 @@ impl ItemRenderer for QtItemRenderer<'_> {
                 }
                 if (last_line_begin < string.size()) {
                     elided = string.left(last_line_begin);
-                    QString to_elide = QStringView(string).mid(last_line_begin, last_line_size).trimmed() % QStringView(u"…");
+                    QString to_elide = QStringView(string).mid(last_line_begin, last_line_size).trimmed() % QStringView(QT_UNICODE_LITERAL("…"));
```
Comment (tronical) [inline sixtyfps_runtime/rendering_backends/qt/qt_window.rs:500]: Neat workaround :)

**Comment by tronical** [comment]
I think our use of `resolver = "2"` in the top-level `Cargo.toml` can also be removed with this?

**Review by tronical — APPROVED** [review]

**Comment by ogoffart** [comment]
Valgrind find the problem in the C++ tests: there are closure that were only using part of the object (in this case the reference to the closure, but not the destructor) so the c++ closure ended up destroyed before it was called. Cargo fix did not see that because the ffi feature was not enabled

File: sixtyfps_runtime/corelib/properties.rs [hunk 2]
```diff
@@ -2387,10 +2387,16 @@ pub(crate) mod ffi {
...
+            fn call(&self) -> i32 {
+                (self.binding)(self.user_data)
+            }
+        }
+
         let c_state_binding = CStateBinding { binding, user_data, drop_user_data };
         let bind_callable = StateInfoBinding {
             dirty_time: Cell::new(None),
-            binding: move || (c_state_binding.binding)(c_state_binding.user_data),
+            binding: move || c_state_binding.call(),
```
Comment (tronical) [inline sixtyfps_runtime/corelib/properties.rs:2399]: Ohh, so here it "moved" just the two fields (`binding` and `user_data`) and then called `drop` on the `c_state`, which called the `x` "destructor"?

**Review by tronical — APPROVED** [review]

File: sixtyfps_runtime/corelib/properties.rs (hunk 2 above)
Comment (ogoffart) [reply sixtyfps_runtime/corelib/properties.rs:2399]: yes.

**Pull request merged**